### DISCUSSION TITLE
Wayland Port using labwc

### DISCUSSION THREAD

**Discussion by ghost** [lxqt/lxqt#1929] [Ideas]
Since lxqt programs are Qt based, those already do run fine on wayland.

lxqt has been labelled as X based DE because, openbox which is used by lxqt is X-exclusive.

https://github.com/johanmalm/labwc is alternative to openbox written using wlroots for Wayland.

Swapping openbox by labwc must ease the port of lxqt to wayland.

What do you think? should we adopt wayland by swapping openbox with labwc ?

**Comment by tsujan** [comment]
LXQt doesn't use and has never used Openbox. Openbox is just one X11 WM among many and, since its development is stopped, it isn't recommended.

As for Wayland, the situation is very different from X11. Although LXQt components should work under all Wayland compositors, we might not be able to remain WM-agnostic anymore. Many possibilities were discussed at https://github.com/lxqt/lxqt/issues/10, most of which weren't modular/good enough and some good options weren't impartial toward Qt and GTK. KWinFT will be the best option if it really becomes independent of KDE.

**Comment by Sunderland93** [comment]
I think the first step towards Wayland (and wlroots-based compositors) is to add support for[ wlr-layer-shell](https://github.com/swaywm/wlr-protocols/blob/master/unstable/wlr-layer-shell-unstable-v1.xml) protocol to lxqt-panel (for example through [qtlayershell](https://github.com/yan12125/qtlayershell) plugin). Note that KWin (and Mir display server) already support that, and therefore LXQt will work great on any wlroots-based compositor and KWin (I'm not sure that KWinFT is support layer-shell). Also likely to need to provide support for [wlr-foreign-toplevel-management](https://github.com/swaywm/wlr-protocols/blob/master/unstable/wlr-foreign-toplevel-management-unstable-v1.xml) protocol, for support windows list.

**Comment by adjamhub** [comment]
I'm following labwc development very closely because I really like that project!
Let's say that during next months I'll work to clean up X11 dependencies on LXQt to try better supporting a wayland WM. I will devote to it my old notebook, installing there an x11-free environment with labwc and trying to let LXQt work there.

**Reply by telmobss1** [reply]
I was here watching the wayfire project but I don't know how it could become the main

**Comment by Sunderland93** [comment]
Just for reference - `layer-shell-qt `project is in KDE umbrella now https://invent.kde.org/plasma/layer-shell-qt

**Reply by adjamhub** [reply]
Many thanks :)
